tgui/packages/tgui/interfaces/Newscaster.js

```diff
@@ -0,0 +1,615 @@
+                  <>
+                    By: {message.auth} at {message.time}
+                  </>
+                )}
+              </i>
+            )}
+            buttons={(
+              <>
+                {!!security_mode && (
+                  <Button
+                    icon="comment-slash"
+                    tooltip="Censor Story"
+                    disabled={!security_mode}
+                    onClick={() => act('storyCensor', {
+                      secure: security_mode,
+                      messageID: message.ID,
+                    })} />
+                )}
+                {!!security_mode && (
+                  <Button
+                    icon="user-slash"
+                    tooltip="Censor Author"
+                    disabled={!security_mode}
+                    onClick={() => act('authorCensor', {
+                      secure: security_mode,
+                      messageID: message.ID,
+                    })} />
+                )}
+                <Button
+                  icon="comment"
+                  tooltip="Leave a Comment."
+                  disabled={
+                    message.censored_author
+                    || message.censored_message
+                    || user.name === "Unknown"
+                    || !!channelLocked && (channelAuthor !== user.name)
+                  }
+                  onClick={() => act('startComment', {
+                    messageID: message.ID,
+                  })}
+                />
+              </>
+            )} >
+            <BlockQuote>
+              {message.censored_message ? (
+                <Section textColor="red">
+                  This message was deemed dangerous to the general welfare
+                  of the station and therefore marked with a <b>D-Notice</b>.
+                </Section>
+              ):(
+                <Section
+                  dangerouslySetInnerHTML={processedText(message.body)}
+                  pl={1} />
+              )}
+              {message.photo !== null && !message.censored_message &&(
+                <Box
+                  as="img"
+                  src={message.photo} />
+              )}
+              {!!message.comments &&(
+                <Box>
+                  {message.comments.map(comment => (
+                    <BlockQuote
+                      key={comment.index}>
+                      <Box
+                        italic
+                        textColor="white">
+                        By: {comment.auth} at {comment.time}
+                      </Box>
+                      <Section
+                        dangerouslySetInnerHTML={processedText(comment.body)}
+                        ml={2.5} />
+                    </BlockQuote>
+                  ))}
+                </Box>
+              )}
+            </BlockQuote>
+            <Divider />
+          </Section>
+        ); }
+      )}
+    </Section>
+  );
+};
+
+export const deconvertHtmlString = (text) => {
```

Comment:
We might already have this function in `common/string` module, decodeHtmlEntities iirc.